src/cashlib/cashlib.cpp

```diff
@@ -0,0 +1,204 @@
+#include <openssl/rand.h>
```

Comment:
can you add the copywrite to both cashlib and sigcommon

+# Copyright (c) 2015-2018 The Bitcoin Unlimited developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.